src/dms/clis/EdFi.DataManagementService.SchemaGenerator.Tests.Unit/Engines/Mssql/__snapshots__/MssqlFlatteningMetadataComprehensiveTests.ValidateNestedCommonCollections_GeneratesMultilevelChildTables.snap

```diff
@@ -0,0 +1,204 @@
+GO
+
+IF NOT EXISTS (SELECT * FROM sys.foreign_keys WHERE name = 'FK_StudentEducationOrganizationAssociationAddressPeriod_Document')
+BEGIN
+    ALTER TABLE [dms].[edfi_StudentEducationOrganizationAssociationAddressPeriod] ADD CONSTRAINT [FK_StudentEducationOrganizationAssociationAddressPeriod_Document] FOREIGN KEY ([Document_Id], [Document_PartitionKey]) REFERENCES [dms].[Document]([Id], [DocumentPartitionKey]);
+    PRINT 'Foreign key constraint FK_StudentEducationOrganizationAssociationAddressPeriod_Document created.';
+END
+ELSE
+    PRINT 'Foreign key constraint FK_StudentEducationOrganizationAssociationAddressPeriod_Document already exists, skipped.';
+GO
+
+
+-- Natural Key Resolution Views
+
+-- Natural Key Resolution View for StudentEducationOrganizationAssociation_View
+IF NOT EXISTS (SELECT * FROM sys.views WHERE name = 'StudentEducationOrganizationAssociation_View' AND SCHEMA_NAME(schema_id) = 'dms')
+BEGIN
+    EXEC('CREATE VIEW [dms].[StudentEducationOrganizationAssociation_View] AS
+    SELECT
+        base.Id,
+        base.Document_Id,
+        base.Document_PartitionKey,
+        base.StudentId
+    FROM [dms].[edfi_StudentEducationOrganizationAssociation] base');
+    PRINT 'View StudentEducationOrganizationAssociation_View created.';
+END
+ELSE
+    PRINT 'View StudentEducationOrganizationAssociation_View already exists, skipped.';
+GO
+
+
+
+-- Natural Key Resolution View for StudentEducationOrganizationAssociationAddress_View
+IF NOT EXISTS (SELECT * FROM sys.views WHERE name = 'StudentEducationOrganizationAssociationAddress_View' AND SCHEMA_NAME(schema_id) = 'dms')
+BEGIN
+    EXEC('CREATE VIEW [dms].[StudentEducationOrganizationAssociationAddress_View] AS
+    SELECT
+        base.Id,
+        base.Document_Id,
+        base.Document_PartitionKey,
+        parent.StudentId AS StudentEducationOrganizationAssociation_StudentId,
+        base.StreetNumberName
+    FROM [dms].[edfi_StudentEducationOrganizationAssociationAddress] base
+    INNER JOIN [dms].[edfi_StudentEducationOrganizationAssociation] parent ON base.StudentEducationOrganizationAssociation_Id = parent.Id');
+    PRINT 'View StudentEducationOrganizationAssociationAddress_View created.';
+END
+ELSE
+    PRINT 'View StudentEducationOrganizationAssociationAddress_View already exists, skipped.';
+GO
+
+
+
+-- Natural Key Resolution View for StudentEducationOrganizationAssociationAddressPeriod_View
+IF NOT EXISTS (SELECT * FROM sys.views WHERE name = 'StudentEducationOrganizationAssociationAddressPeriod_View' AND SCHEMA_NAME(schema_id) = 'dms')
+BEGIN
+    EXEC('CREATE VIEW [dms].[StudentEducationOrganizationAssociationAddressPeriod_View] AS
+    SELECT
+        base.Id,
+        base.Document_Id,
+        base.Document_PartitionKey,
+        ancestor_1.StudentId AS StudentEducationOrganizationAssociation_StudentId,
```

Comment:
I'm assuming the table prefix on the column name is to avoid duplicate column names, but we are basically recreating key unification here. If for some reason there are multiple columns with the same name, just note it with an info-level log message and skip it. They all must collapse to the same column.